File: my-club-web/src/model/team.ts
import {Member} from "./member";
import {Section} from "./section";

export class Team {
  constructor() {
    this.name = ""
    this.description = ""
    return this;
  }

  id: number;
  clubSectionId: Section;

  name: string;
  description: string;

  members: Member[];
}

export const teamsPath: string = "/api/team/"

File: my-club-api/src/main/java/com/my/club/model/Team.java
package com.my.club.model;

import io.quarkus.hibernate.orm.panache.PanacheEntity;
import io.quarkus.hibernate.orm.panache.PanacheQuery;
import jakarta.persistence.Entity;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.ManyToOne;
import java.util.List;

@Entity
public class Team extends PanacheEntity {

    public String name;

    public String description;

    @ManyToOne
    public Section clubSection;

    @ManyToMany
    public List<Member> members;

    public static List<Team> findTeamsBySectionId(long clubSectionId) {
        PanacheQuery<Team> teams = find("clubSection.id=?1", clubSectionId);
        return teams.list();
    }

}
